1xx-enable-ssl-for-kafka-bridge.md

```diff
@@ -0,0 +1,155 @@
+
+`http.ssl.enabled.cipher.suites` can be configured with a comma separated list of cipher suites that the server will support. If not set, the default list of cipher suites provided by the underlying JDK SSL/TLS engine will be used.
+
+If the existing `http.port` is not set and `http.ssl.enable` is set to `true`, the server port will be set to `443` by default. This will allow users to make TLS encrypted connections from their clients for all the HTTP Bridge endpoints for client operations.
+
+This proposal also adds support for a separate listener for internal endpoints such as:
+- `/healthy`
+- `/ready`
+- `/metrics`
+
+The internal endpoints will use the port defined by the new configuration property `http.management.port`, defaulting to 8081 if unspecified. These endpoints will no longer be accessible via the default listener port defined by `http.port` or port 8080. All other endpoints will continue to use the existing listener on port 8080, or 443 if SSL is enabled.
+
+This brings a clear separation of internal and external use for the endpoints and makes it simpler to enable SSL for the listener that is used by external clients. It should also improve performance and security because admin operations, such as collecting metrics, are isolated from traffic for Kafka operations.
+
+Once SSL is enabled, HTTP Bridge server will reject unencrypted connections from external clients. To support both HTTP and HTTPS connections, the recommendation would be to deploy two separate HTTP Bridge instances: one with SSL enabled, and one without. That way external clients can make both encrypted and unencrypted connections for Kafka operations.
+
+### HTTP Bridge managed by Strimzi operator
+
+When deploying `KafkaBridge` custom resources with the Strimzi operator, users can set these configurations in the `spec.http` section of their `KafkaBridge` CR. Users can enable SSL by setting the new field `spec.http.sslEnable`, which is set to `false` by default. When it is set to `true`, users have to provide the certificate and key in PEM format using a Kubernetes `Secret` that is referenced in the `certificateAndKey` field.
+
+Let's take a look at an example of the `KafkaBridge` CR with the new fields to enable SSL:
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaBridge
+metadata:
+  name: my-bridge
+spec:
+  replicas: 3
+  bootstrapServers: <cluster_name>-cluster-kafka-bootstrap:9092
+  http:
+    port: 8443
+    tls: #NEW FIELD
+      serverCertChainAndKey: #NEW FIELD
+        secretName: my-secret
+        certificate: public.crt
+        key: private.key
+...
+```
+The liveness and readiness probes will connect to port 8081 (instead of `http.port` or 8080). If users have metrics set up, they would need to use port 8081 to connect to the `metrics` endpoint and cannot reconfigure this port number as it's only for internal use.
```

Comment:
So we don't allow to change the management port within the `KafkaBridge` custom resource? Why?